nucypher/cli/painting/staking.py

```diff
@@ -267,3 +274,49 @@ def paint_min_rate(emitter, staker):
     ~ Previously set ....... {prettify_eth_amount(raw_minimum)}
     ~ Effective ............ {prettify_eth_amount(minimum)}"""
     emitter.echo(rate_payload)
+
+
+def paint_staking_rewards(stakeholder, blockchain, emitter, past_periods, staking_address, staking_agent):
+    if not past_periods:
+        reward_amount = stakeholder.staker.calculate_staking_reward().to_tokens()
+        emitter.echo(message=TOKEN_REWARD_CURRENT.format(reward_amount=reward_amount))
+        return
+
+    economics = stakeholder.staker.economics
+    seconds_per_period = economics.seconds_per_period
+    current_period = staking_agent.get_current_period()
+    from_period = current_period - past_periods
+    latest_block = blockchain.client.block_number
+    from_block = estimate_block_number_for_period(period=from_period,
+                                                  seconds_per_period=seconds_per_period,
+                                                  latest_block=latest_block)
+
+    argument_filters = {'staker': staking_address}
+    event_type = staking_agent.contract.events['Minted']
+    entries = event_type.getLogs(fromBlock=from_block,
+                                 toBlock='latest',
+                                 argument_filters=argument_filters)
+
+    rows = []
+    rewards_total = 0
+    for event_record in entries:
+        token_reward = NU(event_record['args']['value'], 'NuNit').to_tokens()
+        period = event_record['args']['period']
+        date = datetime_at_period(period, seconds_per_period, start_of_period=True)
+        rows.append([
+            date.local_datetime().strftime("%b %d %Y"),
+            int(event_record['block_number']),
+            int(period),
+            token_reward,
+        ])
+        rewards_total += token_reward
+
+    if not rows:
+        emitter.echo(TOKEN_REWARD_NOT_FOUND)
+        return
+
+    periods_as_days = math.floor(economics.days_per_period * past_periods)
+    emitter.echo(message=TOKEN_REWARD_PAST_HEADER.format(periods=past_periods, days=periods_as_days))
+    emitter.echo(tabulate.tabulate(rows, headers=REWARDS_TABLE_COLUMNS, tablefmt="fancy_grid"))
+    rewards_total = NU(rewards_total, 'NU').to_tokens()
```

Comment:
you don't have to convert NU object to a decimal, it can be represented as a string, and takes care of the details for you eg.
```python
In [1]: from nucypher.blockchain.eth.token import NU

In [2]: n = NU(1_000_000, 'NU')

In [3]: print(f"Total rewards: {n}")
Total rewards: 1000000 NU
```

You can save yourself from calling `to_tokens()` and remove the `NU` text in `TOKEN_REWARD_PAST` string.

Reply:
Same comment/idea for `TOKEN_REWARD_CURRENT` higher up in the file.

Reply:
and incorporate `round` as @cygnusv indicated. 

```python
In [9]: print(f"Total rewards: {round(n, 2)}")
Total rewards: 1000.23 NU

In [10]: n = NU(1000.23132132313, 'NU')

In [11]: print(f"Total rewards: {n}")
Total rewards: 1000.23132132313 NU

In [12]: print(f"Total rewards: {round(n, 2)}")
Total rewards: 1000.23 NU

In [13]: print(f"Total rewards: {round(n, 5)}")
Total rewards: 1000.23132 NU
```